src/IterableCodeExtractor.php

```diff
@@ -38,8 +39,18 @@ public static function fromFile( $file_or_files, Translations $translations, arr
 				}
 			}
 
+			$relative_file_path = ltrim( str_replace( static::$dir, '', Utils\normalize_path( $file ) ), '/' );
+
 			// Make sure a relative file path is added as a comment.
-			$options['file'] = ltrim( str_replace( static::$dir, '', Utils\normalize_path( $file ) ), '/' );
+			$options['file'] = $relative_file_path;
+
+			if ( ! empty( $options['restrictDirectories'] ) ) {
+				$top_level_dirname = explode( '/', $relative_file_path )[0];
```

Comment:
This only considers the top-level `styles` folder of a theme — provided the root directory for the command is the theme folder. Is there any kind of support for themes that are nested within each other? Example: `wp-content/themes/my-theme-generator/my-child/`. There can be theme.json or style variations both `my-theme-generator` and `my-theme-generator/my-child`. I suppose the theme directory only admits "single themes", not "nested ones", so this wouldn't be an issue for string extraction.

Reply:
> Is there any kind of support for themes that are nested within each other? Example: `wp-content/themes/my-theme-generator/my-child/`

Is that even a thing? 🤔 

If it's a nested theme that can be activated on its own, then it needs its own text domain anyway, so you wouldextract the strings for `my-child` individually, with `my-child` being the root.

> I suppose the theme directory only admits "single themes", not "nested ones", so this wouldn't be an issue for string extraction.

Correct.